cub/device/dispatch/dispatch_segmented_sort.cuh

```diff
@@ -1653,7 +1658,7 @@ private:
             return error;
         }
 
-        if (CubDebug(error = SyncStream(stream)))
+        if (CubDebug(error = DebugSyncStream(stream)))
```

Comment:
This sync is **not** meant as debug one. It's required for correctness, which contradicts the documentation of the `DebugSyncStream`. While reading the rest of the PR, I was thinking that `DebugSyncStream` usage is a bit bloated and unsafe. It's impossible to guarantee that the `DebugSyncStream` is called inside `if(debug)`. In order to guarantee this, can we change the signature to something like `DebugSyncStream(stream, debug)` and hide the branch inside? This would address mentioned concerns.

Reply:
Good suggestion. Switched this to a regular `SyncStream` and refactored `DebugSyncStream` to take a `debug_synchronous` argument.